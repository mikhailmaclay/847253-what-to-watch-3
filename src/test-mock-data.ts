import {User, Movie, Review, TeamMember, Notification} from './types';

export const mockUser: User = {
  id: 1,
  email: `JohnnyMnemonyc@mail.ru`,
  name: `Johnny`,
  avatar: `https://htmlacademy-react-3.appspot.com/wtw/static/avatar/4.jpg`,
  myList: null
};

export const mockReview: Review = {
  id: 1,
  user: {id: mockUser.id, fullName: mockUser.name},
  date: `2020-03-24T08:26:50.139Z`,
  rating: 10,
  text: `It was really awesome!`
};

export const mockDirector: TeamMember = {
  fullName: `Wes Anderson`,
  role: `Director`
};

export const mockActor: TeamMember = {
  fullName: `Adrien Brody`,
  role: `Actor`
};

export const mockMovie: Movie = {
  id: 1,
  name: `Dardjeeling Limited`,
  genre: `Adventure`,
  releaseDate: `2007`,
  description: [
    `A year after their father\`s funeral, three brothers travel across India by train in an attempt to bond with each other.`
  ],
  runTime: 91,
  team: [
    mockDirector,
    mockActor
  ],
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Dardjeeling_Limited.jpg`,
  images: [
    `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/dardjeeling_limited.jpg`
  ],
  backgrounds: [
    `#AD9F8B`,
    `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Dardjeeling_Limited.jpg`
  ],
  rating: 3.6,
  scoresCount: 165106,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  isInMyList: false,
  reviews: null
};

export const mockNotification: Notification = {
  title: `Test`,
  content: `Testing message.`
};
