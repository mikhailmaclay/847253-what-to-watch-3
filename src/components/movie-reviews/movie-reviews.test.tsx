// Libraries
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// Components
import MovieReviews from './movie-reviews';
//
import {mockReview} from '../../test-mock-data';

describe(`<MovieReviews/>`, () => {
  const props = {
    reviews: [mockReview]
  };

  it(`Should render correctly`, () => {
    const result = renderer.create(
        <MovieReviews {...props}/>
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});
