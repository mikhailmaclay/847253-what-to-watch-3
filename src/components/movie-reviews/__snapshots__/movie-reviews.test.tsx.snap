// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieReviews/> Should render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          It was really awesome!
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Johnny
          </cite>
          <time
            className="review__date"
            dateTime="2020-03-24T08:26:50.139Z"
          >
            March 24, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        10.0
      </div>
    </div>
  </div>
</div>
`;
